packages/core/src/useForm/useForm.spec.ts

```diff
@@ -267,6 +267,49 @@ describe('form reset', () => {
     expect(isTouched('foo')).toBe(true);
   });
 
+  test('can reset form path values and touched to their original state', async () => {
+    const { values, reset, setValue, isTouched, setTouched } = await renderSetup(() => {
+      return useForm({
+        initialValues: { company: { employee: { name: 'John' } } },
+        initialTouched: { company: { employee: { name: true } } },
+      });
+    });
+
+    setValue('company.employee.name', 'Alice');
+    setTouched('company.employee.name', false);
+
+    expect(values).toEqual({ company: { employee: { name: 'Alice' } } });
+    expect(isTouched('company.employee.name')).toBe(false);
+    reset('company.employee.name');
+
+    expect(values).toEqual({ company: { employee: { name: 'John' } } });
+    expect(isTouched('company.employee.name')).toBe(true);
+  });
+
+  test('can reset from path values and touched to a new state', async () => {
+    const { values, reset, setValue, isTouched, setTouched } = await renderSetup(() => {
+      return useForm({
+        initialValues: { company: { employee: { name: 'John' } } },
+      });
+    });
+
+    reset('company.employee.name', {
+      value: 'Alice',
+      touched: true,
+    });
+
+    expect(values).toEqual({ company: { employee: { name: 'Alice' } } });
+    expect(isTouched('company.employee.name')).toBe(true);
+
+    setTouched('company.employee.name', false);
+    setValue('company.employee.name', 'Bob');
+
+    reset('company.employee.name');
+
+    expect(values).toEqual({ company: { employee: { name: 'Alice' } } });
+    // expect(isTouched('company.employee.name')).toBe(true);
```

Comment:
We still need a way to set the original touched. Currently `updateTouchedPath` works to touch values, I think its not what we want when we're resetting.

Reply:
Right! I added `setInitialTouchedPath` just now, do we need similar one for dirty?

Reply:
> Right! I added `setInitialTouchedPath` just now, do we need similar one for dirty?

I don't think so because we can already do `form.revertDirty(path)`, and also when we reset a form/path, we can't pass a new `dirty` state -- only new values and touched state.